kroxylicious-filters/kroxylicious-authorization/src/main/java/io/kroxylicious/filter/authorization/OffsetFetchGroupBatchingEnforcement.java

```diff
@@ -51,13 +54,13 @@ CompletionStage<RequestFilterResult> onRequest(RequestHeaderData header,
                                                    FilterContext context,
                                                    AuthorizationFilter authorizationFilter) {
         List<Action> actions = request.groups().stream()
-                .flatMap(ofrq -> ofrq.topics().stream())
+                .flatMap(ofrq -> Optional.ofNullable(ofrq.topics()).stream().flatMap(Collection::stream))
```

Comment:
```
    { "name": "Topics", "type": "[]OffsetFetchRequestTopic", "versions": "0-7", "nullableVersions": "2-7",
      "about": "Each topic we would like to fetch offsets for, or null to fetch offsets for all topics.", "fields": [
```

This means the response will contain info about all the topics, potentially including topics the subject is not authorized to see. So the correct fix requires handling the all topics case by intercepting the response and filtering-out any topics which the subject is not authorized to describe.